src/v/cluster/metadata_cache.cc

```diff
@@ -111,9 +111,9 @@ ss::future<std::vector<broker_ptr>> metadata_cache::all_alive_brokers() const {
 
     std::set<model::node_id> brokers_with_health;
     for (auto& st : res.value()) {
+        brokers_with_health.insert(st.id);
         if (st.is_alive) {
             auto broker = _members_table.local().get_broker(st.id);
-            brokers_with_health.insert(st.id);
             if (broker) {
                 brokers.push_back(std::move(*broker));
             }
```

Comment:
I'm having trouble reconciling what the commit message says with the code.

For example:

```
When any of the broker was down the logic automatically returned full
brokers list.
```

It would be helpful to reconcile the logic if the commit message were disambiguated with respect to returning all brokers because `brokers` were empty at the end of the method or because `brokers` contained all brokers (e.g. if `brokers_with_health` were empty in the corner case or I think if all the brokers were not alive and weren't in the members table?).

Reply:
I am not sure if i understood the suggestion correctly. I've added more detailed commit message and included the empty `brokers` list case in the explanation.

Reply:
thx for the extra context it makes sense now